### DISCUSSION TITLE
createContext only works in Client Components. Add the "use client" in package

### DISCUSSION THREAD

**Discussion by i-ked-ii** [vercel/next.js#70656] [Help]
### Summary

![Screenshot 2567-10-01 at 11 23 38](https://github.com/user-attachments/assets/2668c69f-9a37-4752-867f-bcd7fa0ecc7a)

I have problem with "use client" in package `rc-field-form` . Next.js must be check other package?
If I install other not for next.js I will have error form "use client" or not?

How I can fix this problem.

### Additional information

```js
next info

Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.6.0: Mon Jul 29 21:16:46 PDT 2024; root:xnu-10063.141.2~1/RELEASE_ARM64_T8112
  Available memory (MB): 24576
  Available CPU cores: 8
Binaries:
  Node: 21.7.1
  npm: 10.8.3
  Yarn: 1.22.19
  pnpm: N/A
Relevant Packages:
  next: 14.2.13 // Latest available version is detected (14.2.13).
  eslint-config-next: N/A
  react: 18.3.1
  react-dom: 18.3.1
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

**Comment by GabenGar** [comment]
By creating an issue in their repo.
The [file in question](https://github.com/react-component/field-form/blob/master/src/FieldContext.ts) is not written with React Server Components in mind, so that's the upstream issue.

**Comment by icyJoseph** [comment]
I think you can do:

```ts
// lib/rc-field-form-client.ts or whatever name you please
'use client';

export * as default from 'rc-field-form';
```

Or something like that, I don't remember the syntax, and then consume from that file instead of `rc-field-form` directly. And then you wait for `rc-field-form`, or React to come up with a solution.
